### PR TITLE
Add Python 3.8 support

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Adds Python 3.8 support to `tox.ini` and `.circleci/config.yml`.

Also changes the docker image to use the one published to https://github.com/DataDog/dd-trace-ci.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,7 +7,10 @@ httpbin_local: &httpbin_local
   image: kennethreitz/httpbin@sha256:2c7abc4803080c22928265744410173b6fea3b898872c01c5fd0f0f9df4a59fb
   name: httpbin.org
 test_runner: &test_runner
-  image: datadog/docker-library:ddtrace_py
+  image: docker.pkg.github.com/datadog/dd-trace-ci/ddtracepy-dev:1.0
```
Comment (brettlangdon) [inline .circleci/config.yml]: Just leaving a note since we wanted to update this: `dd-trace-ci/python-dev:latest`

File: docs/advanced_usage.rst [hunk 2]
```diff
@@ -281,7 +281,7 @@ Where environment variables are not used for configuring the tracer, the instruc
 
 Resolving deprecation warnings
 ------------------------------
-Before upgrading, it’s a good idea to resolve any deprecation warnings raised by your project.
+Before upgrading, it's a good idea to resolve any deprecation warnings raised by your project.
```
Comment (jd) [inline docs/advanced_usage.rst]: ?

File: tox.ini [hunk 3]
```diff
@@ -70,67 +70,72 @@ envlist =
...
-    unit_tests-{py27,py34,py35,py36,py37}
-    benchmarks-{py27,py34,py35,py36,py37}
+    unit_tests-{py27,py34,py35,py36,py37,py38}
+    benchmarks-{py27,py34,py35,py36,py37,py38}
 
 [testenv]
+# Newer versions of Tox have issues decoding unicode output in Python 2.7
+# https://github.com/tox-dev/tox/issues/1369
+setenv =
+    LC_ALL=en_US.UTF-8
```
Comment (jd) [inline tox.ini]: that seems unrelated to the PR and not something I encountered anywhere yet?
it'd be good to have this in a different PR since it's a different topic.

File: docs/advanced_usage.rst (hunk 2 above)
Comment (Kyle-Verhoog) [reply docs/advanced_usage.rst]: I was trying to find random unicode characters floating around the repo because of unicode errors with the docs and found this one.

It didn't fix anything but it doesn't need to be unicode anyway  🤷‍♂

Comment (jd) [reply docs/advanced_usage.rst]: Alright, I don't like seeing things unrelated to the PR subject being changed.
You're good for once. ;)

File: tox.ini [hunk 4]
```diff
@@ -72,67 +73,70 @@ envlist =
...
+    py{37,38}-opentracer_gevent-gevent{13,14}
 # Unit tests: pytest based test suite that do not require any additional dependency
-    unit_tests-{py27,py34,py35,py36,py37}
-    benchmarks-{py27,py34,py35,py36,py37}
+    unit_tests-{py27,py34,py35,py36,py37,py38}
+    benchmarks-{py27,py34,py35,py36,py37,py38}
 
 [testenv]
+# Newer versions of Tox have issues decoding unicode output in Python 2.7
+# https://github.com/tox-dev/tox/issues/1369
```
Comment (jd) [inline tox.ini]: I'm confused about the link with 3.8 here?

Comment (Kyle-Verhoog) [reply tox.ini]: Yeah there is none - this was left in from me debugging the unicode issues (which were tox related). But it was resolved using Python 3 to run tox so this comment can go away.

**Comment by jd** [comment]
@Kyle-Verhoog  I'm taking over this PR so we can merge it soon 😊

File: tox.ini [hunk 5]
```diff
@@ -880,6 +934,9 @@ setenv =
 setenv =
     {[bottle_autopatch]setenv}
 [testenv:bottle_contrib_autopatch-py37-bottle12-webtest]
+setenv =
+    {[bottle_autopatch]setenv}
+[testenv:bottle_contrib_autopatch-py38-bottle12-webtest]
```
Comment (brettlangdon) [inline tox.ini]: I hate all of these...
